src/main/java/edu/wpi/grip/core/SocketHint.java

```diff
@@ -56,6 +69,14 @@ public SocketHint(String identifier, Class<T> type, T initialValue) {
         this(identifier, type, () -> initialValue);
     }
 
+    protected void setPublishable(boolean publishable) {
```

Comment:
Having a method exclusively for unit tests isn't really a good idea. If you REALLY, (this doesn't seem like a valid reason to use reflection) want to run a method in a unit test that should be private use reflection to run it.

Also, the unit test has access to the constructor too.